examples/src/main/java/com/thousandeyes/sdk/PaginatorExample.java

```diff
@@ -0,0 +1,48 @@
+package com.thousandeyes.sdk;
+
+import java.util.List;
+
+import com.thousandeyes.sdk.client.ApiClient;
+import com.thousandeyes.sdk.client.ApiException;
+import com.thousandeyes.sdk.client.NativeApiClient;
+import com.thousandeyes.sdk.endpoint.agents.ManageAgentsApi;
+import com.thousandeyes.sdk.endpoint.agents.model.EndpointAgent;
+import com.thousandeyes.sdk.endpoint.agents.model.ListEndpointAgentsResponse;
+import com.thousandeyes.sdk.pagination.Paginator;
+
+
+
+class PaginatorExample {
+    private static final ApiClient apiClient = NativeApiClient
+            .builder()
+            .baseUri("https://api.stg.thousandeyes.com/v7")
+            .bearerToken("<token>")
+            .build();
+
+    public static void main(String[] args) {
+        var apiInstance = new ManageAgentsApi(apiClient);
+        var aid = "11";
+        var max = 1;
+
+        Paginator<EndpointAgent, ListEndpointAgentsResponse> paginator = new Paginator<>(
+                cursor -> {
+                    try {
+                        return apiInstance.getEndpointAgents(max, cursor, aid, List.of(), false,
+                                                             false, null, "CSCO-W-PW02XZDB");
+                    }
+                    catch (ApiException e) {
+                        throw new RuntimeException(e);
+                    }
+                },
+                ListEndpointAgentsResponse::getAgents
```

Comment:
ideally we would abstract away this logic. In the scope of the spike I didn't invest much time in figuring out how

Reply:
agree yeah, should be simple since we control 100% the SDK generation logic we can add a new method that already returns this `Paginator` instance.

Reply:
@brumarqu-te, just curious how you imagine this working? Because it needs to be at method/operation level, rather than API level, right? So are you thinking we will use some custom [extension](https://swagger.io/docs/specification/openapi-extensions/) to define which operations are paginated, and what the primary field is?

Or maybe I'm totally missing something 🤔

Reply:
i imagined those being conditionally added as an additional operation during the post processing phase of the generator. haven't put much thought into what the condition would be but real quick and dirty: introspecting the input for `cursor` or output for `_links.next` ...

... maybe `simple` was not the best word 😅 but definitely "doable" 😄